File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxCandies = max(candies)
        result = []
        for i in range(len(candies)):
            if candies[i]+extraCandies >= maxCandies:
                result.append(True)
            else:
                result.append(False)
        return result

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> bigCandiesList = new ArrayList<>();

        int maxCandiesNum = 0;

        for(int candie : candies){
            maxCandiesNum = Math.max(maxCandiesNum,candie);
        }

        for(int candie : candies){
            if (candie + extraCandies >= maxCandiesNum){
                bigCandiesList.add(true);
            }else{
                bigCandiesList.add(false);
            }
        }
        return bigCandiesList;
    }
}
